x-pack/filebeat/input/awss3/sqs_s3_event.go

```diff
@@ -252,6 +285,17 @@ func (p *sqsS3EventProcessor) getS3Notifications(body string) ([]s3EventV2, erro
 		}
 	}
 
+	// Check if the notification is from S3 -> EventBridge -> SQS
+	if events.Records == nil {
```

Comment:
We are also checking `if events.Records == nil` in line 299 and return error. Should we remove that?

Reply:
@kaiyan-sheng The decode / convert should populate the events.Records and if it's not nil then it was able to parse (and thus recieved) the 'normal' S3 event. 

I agree it's not the prettiest thing, am open to suggestions. :-)